modules/schema/src/main/java/org/apache/ignite/internal/schema/registry/SchemaRegistryImpl.java

```diff
@@ -145,6 +181,7 @@ public void onSchemaDropped(int ver) {
         if (ver >= lastVer || ver <= 0 || schemaCache.keySet().first() < ver)
             throw new SchemaRegistryException("Incorrect schema version to clean up to: " + ver);
 
-        schemaCache.remove(ver);
+        if (schemaCache.remove(ver) != null)
+            mappingCache.keySet().stream().filter(k -> (k & 0xFFFFFFFF) == ver).forEach(mappingCache::remove);
```

Comment:
```suggestion
           mappingCache.keySet().removeIf(k -> (k & 0xFFFFFFFFL) == ver);
```

Please note the `L` at the end of the integer literal. Without it cache won't be cleaned.